### ISSUE TITLE
When localshop not found a version great than the last stored in database start an update action.

### ISSUE THREAD

**Issue by wiliamsouza** [mvantellingen/localshop#165]
When localshop not found a version great than the last stored in database start an update action.
For example I have `py==1.4.28` in localshop db and the last version of`pypi` is `py==1.4.29`.
To force the update I need interact with Django admin and remove `py` db entry.
What is the best solution to this?

**Comment by canassa** [comment]
I already had that problem before, this happens when you are trying to get a package version that was not fetched from the Celery beat task that runs every night.

There is a "update" button on the package page which should force an update from PyPI, you can try using that to force an update from PyPI.

**Comment by wiliamsouza** [comment]
What about change a little bit how localshop works with not found version to try fetch from `pypi` before return an error? Is it worth?

**Comment by canassa** [comment]
I think that if the package is not local, we could return a redirect to the PyPI server and start an background task to fetch the new package. What do you think about this @mvantellingen ?

**Comment by mvantellingen** [comment]
Yeah I agree, If a specific version is asked which is not known yet then we need to do an update.

**Comment by canassa** [comment]
@wiliamsouza

What about sending us a pull request? :smiley:

**Comment by edersonbadeca** [comment]
:+1:

**Comment by wiliamsouza** [comment]
@canassa Challenge accepted :D

**Comment by bsdlp** [comment]
@canassa wrt https://github.com/mvantellingen/localshop/issues/165#issuecomment-114813573 is clicking the refresh button confirmed to sync versions available from upstream? i.e. gevent 1.1.1 release today after the nightly job but clicking refresh does not make 1.1.1 available to localshop.
